dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventAnalyticsService.java

```diff
@@ -390,13 +392,35 @@ private Grid generateOutputGrid( Grid grid, EventQueryParams params,
                 fillDisplayList = false;
             }
 
-            rowDimensions.forEach( dimension -> outputGrid
-                .addValue( displayObjects.get( dimension ).getDisplayProperty( params.getDisplayProperty() ) ) );
+            addValuesInOutputGrid( rowDimensions, outputGrid, displayObjects, params );
 
             EventAnalyticsUtils.addValues( ids, grid, outputGrid );
         }
 
-        return outputGrid;
+        return getGridWithRows( grid, outputGrid );
+    }
+
+    /**
+     * return valid grid. Valid grid is first output output grid with rows or
+     * the basic one
+     */
+    private static Grid getGridWithRows( Grid grid, Grid outputGrid )
+    {
+        return outputGrid.getRows().isEmpty() ? grid : outputGrid;
+    }
+
+    /**
+     * add values in output grid. Display objects are not empty if columns and
+     * rows are not epmty
+     */
+    private static void addValuesInOutputGrid( List<String> rowDimensions, Grid outputGrid,
```

Comment:
As this may or not add values to the `outputGrid`, I think that this method could be named "maybeAddValuesInOutputGrid". Because the values will be added only if `!displayObjects.isEmpty()`

Reply:
Done